src/gui/src/ActionDialog.h

```diff
@@ -20,36 +20,38 @@
 
 #include <QDialog>
 
-#include "ui_ActionDialogBase.h"
-
 class Hotkey;
 class Action;
 class QRadioButton;
 class QButtonGroup;
 class ServerConfig;
+class KeySequenceWidget;
+
+namespace Ui {
+class ActionDialog;
+}
 
-class ActionDialog : public QDialog, public Ui::ActionDialogBase
+class ActionDialog : public QDialog
 {
   Q_OBJECT
 
 public:
   ActionDialog(QWidget *parent, ServerConfig &config, Hotkey &hotkey, Action &action);
+  ~ActionDialog() override;
 
 protected slots:
-  void accept();
+  void accept() override;
   void on_m_pKeySequenceWidgetHotkey_keySequenceChanged();
 
 protected:
-  const KeySequenceWidget *sequenceWidget() const
-  {
-    return m_pKeySequenceWidgetHotkey;
-  }
+  const KeySequenceWidget *sequenceWidget() const;
   const ServerConfig &serverConfig() const
   {
     return m_ServerConfig;
   }
 
 private:
+  std::unique_ptr<Ui::ActionDialog> ui_;
```

Comment:
```suggestion
  std::unique_ptr<Ui::ActionDialog> m_ui;
```

Please follow the code style and use `m_` style members: https://github.com/deskflow/deskflow/wiki/Code-Style#naming

Reply:
id rather not . we have limited space and extra chars are gonna eat my limited line space.

I plan to update the guide line for this naming convention any way 

1. All the object on your form are pointers so adding p is just stupid
2. ALL Qt members with a ui files use the name space `ui` if i am changing `ui_` its to `ui`
3. Limited line space long ass name prefixes =  more poorly linted lines.

Reply:
honestly rather change this to just `ui` since that is the Qt standard for the ui dialog class name.